### ISSUE TITLE
Models stuck together at bulbs

### ISSUE THREAD

**Issue by pearce31** [cboulay/PSMoveService#109]
Release 4.5 move models in steam vr are stuck together at the bulbs

**Comment by HipsterSloth** [comment]
I think this might be related to Issue #94. There was a recent change to SteamVR that affects hand controller roles that I think is causing this issue for some people. Would you mind giving this custom build of psmoveservice a try: https://drive.google.com/open?id=0B5GgUbBXlQ1ua252U3dfdGV6NUk. And make sure to re-install the steamvr driver when you try this custom build out (since the potential fix is in the steamvr plugin not the service).

**Comment by pearce31** [comment]
OK will do

**Comment by zelmon64** [comment]
It could also be a masking problem. If the cyan mask is actually looking at magenta, or visa versa, this would happen. Might be worth a quick check.

**Comment by HipsterSloth** [comment]
Oh that is a great point actually @zelmon64! I think I've seen something like that in the past when the lighting conditions weren't great in the room I was testing in. Enough of the 1st controllers color bled through into the second colors filter that it picked it up the 1st controller's bulb for the second controller.

**Comment by pearce31** [comment]
How do you make sure the cyan mask is looking at cyan vice versa?

**Comment by pearce31** [comment]
I did the calibrations with little light is that good or bad?

**Comment by pearce31** [comment]
So you can't show magenta move when you are using cyan in tool? If so that's why. I showed both at same color.

**Comment by pearce31** [comment]
Are both controllers supposed to be lit when first using tools. Because only the one was lit.

**Comment by zelmon64** [comment]
@pearce31 The colour calibration is only done with the first controller. There are plans to enable the other controllers to ensure there's no cross-talk. For the moment you can decrease the three range values to their minimum suitable tolerance. 

If you didn't know, the short cut to setting the values is to right click on where the PSMove orb is shown on the live camera feed.

**Comment by pearce31** [comment]
Sorry really confused I don't know where the info gets sent when right clicking bulb. Where do I enter the info?

**Comment by zelmon64** [comment]
@pearce31 Have you gone through [this wiki page](https://github.com/cboulay/PSMoveService/wiki/Color-Filter-Calibration)? When right-clicking the orb it sets the values in the `Controller Color` section. The three range values are also in that part of the window. Don't forget to save.

**Comment by pearce31** [comment]
Yes but don't understand the complete process to do on step 4

**Comment by pearce31** [comment]
If its just to only right click the bulb I've done that

**Comment by zelmon64** [comment]
If it's done correctly you should then be able to see the orb with the masked `Video Filter Mode`. You can then manually reduce the range values.

**Comment by pearce31** [comment]
Yes the bulb is filled in in masked mode. But bulbs are stuck together in steam vr inside hmd

**Comment by zelmon64** [comment]
Yes, that is why I am suggesting to minimize the range values. Have you tried that yet?

**Comment by zelmon64** [comment]
Alternatively, if you turn a PSMove off then back on while PSMoveErvice is running, it will reconnect as a different controller number and therefore a different colour.

**Comment by pearce31** [comment]
OK thanks appreciate the help I will try and write back.

**Comment by pearce31** [comment]
theres no issue with the service. i was making a stupid mistake the whole time. ps move service is awesome. now just need osvr tracker camera update to make it work great without the drift and jumpiness.

**Comment by HipsterSloth** [comment]
There are no stupid mistakes here, only valuable lessons to record in the wiki :). If you don't mind me asking, what was the mistake? If it was caused by a usability issue i'd love to hear about it so that we can make the tool easier to use.

**Comment by pearce31** [comment]
when i was adjusting the gain and exposure i thought when everything got to the brightest then went back to the darkest that it was completely looping. i didnt even realise it wasnt. it was just increasing more and more and i was putting the gain and exposure way too high which caused the problem.

**Comment by HipsterSloth** [comment]
Ah! That is definitely a bug on our part. It should loop those values. I will update the issue title and re-label this a bug.

**Comment by pearce31** [comment]
glad to help. great work your doing. thanks so much! keep it up!

**Comment by pearce31** [comment]
ok. so i checked again. and exposure is looping correctly but gain is the one that keeps increasing in values and this is in the 4.5.1 build you posted here from drive.
